contentgrid-appserver-query-engine-impl-jooq/src/test/java/com/contentgrid/appserver/query/engine/jooq/EventsDispatchTest.java

```diff
@@ -0,0 +1,878 @@
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(123)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
+
+        queryEngine.update(
+                APPLICATION,
+                EntityData.builder()
+                        .name(ENTITY_A.getName())
+                        .id(created.getId())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(456)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                updateEventConsumer
+        );
+
+        Mockito.verify(updateEventConsumer).onEntityUpdate(
+                eq(APPLICATION),
+                argThat(oldData -> {
+                    var attribute = oldData.getAttributeByName(ATTRIBUTE_A.getName());
+                    if (attribute.isPresent() && attribute.get() instanceof SimpleAttributeData<?> attr) {
+                        assertThat(attr.getValue()).isEqualTo(123L);
+                        return true;
+                    }
+                    return false;
+                }),
+                argThat(newData -> {
+                    var attribute = newData.getAttributeByName(ATTRIBUTE_A.getName());
+                    if (attribute.isPresent() && attribute.get() instanceof SimpleAttributeData<?> attr) {
+                        assertThat(attr.getValue()).isEqualTo(456L);
+                        return true;
+                    }
+                    return false;
+                })
+        );
+
+        // Transaction completed → it's updated in db
+        var updated = queryEngine.findById(APPLICATION, created.getIdentity().toRequest(), PERMIT_ALWAYS).orElseThrow();
+        var updatedAttr = (SimpleAttributeData<?>) updated.getAttributeByName(ATTRIBUTE_A.getName()).orElseThrow();
+        assertThat(updatedAttr.getValue()).isEqualTo(456L);
+    }
+
+    @Test
+    void verifyDelete_happy() {
+        var created = queryEngine.create(
+                APPLICATION,
+                EntityCreateData.builder()
+                        .entityName(ENTITY_A.getName())
+                        .attribute(SimpleAttributeData.builder()
+                                .name(ATTRIBUTE_A.getName())
+                                .value(123)
+                                .build())
+                        .build(),
+                PERMIT_ALWAYS,
+                createEventConsumer
+        );
+
+        queryEngine.delete(
+                APPLICATION,
+                EntityRequest.forEntity(created.getName(), created.getId()),
```

Comment:
```suggestion
                created.getIdentity().toRequest(),
```

(Unless you explicitly want a delete that's not version-checked, but that doesn't seem to be the case from this context)